src/services/iamRole/data.ts

```diff
@@ -224,14 +228,13 @@ export const listIamRoles = async ({
                       .map(p => p.ManagedPolicies)
                       .reduce((current, acc) => [...acc, ...current], []) || [],
                   Tags: tags.find(t => t?.RoleName === RoleName)?.Tags || {},
-                  PermissionsBoundaryArn:
-                    PermissionsBoundary.PermissionsBoundaryArn,
-                  InlinePolicies: roleAuthorizationDetails
-                    .find(rAD => rAD.RoleName === RoleName)
-                    .RolePolicyList.map(rPl => ({
-                      name: rPl.PolicyName,
-                      document: rPl.PolicyDocument,
-                    })),
+                  ...(PermissionsBoundary?.PermissionsBoundaryArn
```

Comment:
why an object here if we are only storing one property?

Reply:
This small check deconstructs into the property we need to add if the value is not undefined. This way, I don't introduce an empty/undefined property to the result